src/bosh_azure_cpi/lib/cloud/azure/restapi/azure_client.rb

```diff
@@ -2653,12 +2742,13 @@ def http_get_response(uri, request, retry_after)
 
         status_code = response.code.to_i
         response_body = response.body
+        truncated_body = response_body.to_s.length > TRUNCATION_LIMIT ? response_body.to_s[0..TRUNCATION_LIMIT] + '... [truncated]' : response_body.to_s
         message = "http_get_response - #{status_code}\n"
         message += get_http_common_headers(response)
         message += if filter_credential_in_logs(uri)
                      'response.body cannot be logged because it may contain credentials.'
                    else
-                     "response.body: #{redact_credentials_in_response_body(response_body)}"
+                     "response.body: #{redact_credentials_in_response_body(truncated_body)}"
```

Comment:
Was the goal of this truncation to limit the log size, or because the redaction process itself was slow? 

The `redact_credentials_in_response_body` tries to JSON parse the body you pass to it. By truncating the JSON string first, the `redact_credentials_in_response_body` may not be able to parse it, which means it will not redact any of the credentials, leaking credentials. If you redact first, then truncate, you shouldn't have that problem.

Reply:
You've got a point - I will change this when I'm back from traveling (early June). Goal was to truncate, because the logs were unreadable due to the large response bodies.

Reply:
Omitting large responses rather than trying to redact then truncate seems reasonable to me.  👍